src/core/eventhandler.rs

```diff
@@ -111,9 +114,53 @@ impl EventHandler for Handler {
       .await
       .unwrap()
       .is_own(&ctx.cache);
+    let user_id = reaction.user_id.unwrap();
+    let guild = parse::get_guild(reaction.channel_id, &ctx, None)
+      .await
+      .unwrap();
 
-    if reaction.user_id.unwrap() != botid && isown {
+    if user_id != botid && isown {
       let emoji = reaction.emoji.as_data();
+
+      if let Ok(member) = guild.member(&ctx.http, user_id).await {
+        if member.roles.len() >= 1
+          && member
+            .roles
+            .iter()
+            .all(|role| ROLE_AUTHORIZED_TO_DEPLOY.contains(&role.0))
+        {
+          let mut handler = REACTION_COLLECTORS.write().await;
+
+          if let Some(reaction_collector) = handler.get(&reaction.message_id) {
+            if reaction.emoji == reaction_collector.accept {
+              log::info!("ACCEPTED: {}", reaction_collector.short_sha);
+
+              reqwest::Client::new()
+                .post(CODEFLOW_SUPERVISOR_URL.format_url(format!(
```

Comment:
NOTE: we will see later but we probably want "some" security to avoid anyone calling this route.

Reply:
If you want we can put a x-api-key for deployment routes ?